File: CS2242_SMART_WHEELCHAIR/src/system/Person.java
package system;

import java.util.ArrayList;

public class Person {
	private String name;
	private int nic;
	private String address;
	private int age;
	private String gender;
	private CareHouse relatedCareHouse;
	private ArrayList<ContactDetail> contactDetails;
	public Person(String name,int nic,String address,int age, String gender,CareHouse relatedCareHouse,ContactDetail contactDetail) {
		this.name = name;
		this.nic = nic;
		this.address =address;
		this.age =age;
		this.gender = gender;
		this.relatedCareHouse =relatedCareHouse;
		this.contactDetails.add(contactDetail);
	}
	protected String getName() {
		return name;
	}
	protected int getNic() {
		return nic;
	}
	protected String getAddress() {
		return address;
	}
	protected int getAge() {
		return age;
	}
	protected String getGender() {
		return gender;
	}
	protected CareHouse getRelatedCareHouse() {
		return relatedCareHouse;
	}
	protected ArrayList<ContactDetail> getContactDetails() {
		return contactDetails;
	}
	protected void changeAddress(String address) {
		this.address = address;
	}
	protected void changeRelatedCareHouse(CareHouse relatedCareHouse) {
		this.relatedCareHouse = relatedCareHouse;
	}
	protected void addContactDetails(ContactDetail contactDetail) {
		this.contactDetails.add(contactDetail);
	}
	protected void removeContactDetails(ContactDetail contactDetail) {
		this.contactDetails.remove(contactDetail);
	}
	
	
}

File: C++ Classes/person.cpp
#include"person.h"

Person::Person(const std::string& name, const std::string& nic, const std::string& address,
                const int& age, Gender gender, CareHouse* related_care_house)
:name(name), nic(nic), address(address), age(age), gender(gender), related_care_house(related_care_house){
    
}

//setters
void Person::change_address(const std::string& new_address){
    address = new_address;
}
void Person::change_related_care_house(CareHouse& care_house){
    related_care_house = &care_house;
}
void Person::add_contact(contact_type contact_type, const std::string& contact){
    switch(contact_type){
        case(LAND):
            land_contacts.push_back(contact);
            break;
        case(MOBILE):
            mobile_contacts.push_back(contact);
    }
}
void Person::remove_contact(contact_type contact_type, const std::string& contact){
    switch(contact_type){
        case(LAND):
            remove(land_contacts.begin(),land_contacts.end(), contact);
            break;
        case(MOBILE):
            remove(mobile_contacts.begin(), mobile_contacts.end(), contact);
    }
}

//getters
const std::string& Person::get_name() const{
    return name;
}
const std::string& Person::get_nic() const{
    return nic;
}
const std::string& Person::get_address() const{
    return address;
}
const int& Person::get_age() const{
    return age;
}
const int& Person::get_gender() const{
    switch(gender){
        case(MALE):
            return 0;
        case(FEMALE):
            return 1;
    }
}
const CareHouse& Person::get_related_care_house() const{
    return *related_care_house;
}
const std::vector<std::string>& Person::get_contacts(contact_type contact_type) const{
    switch(contact_type){
        case(LAND):
            return land_contacts;
        case(MOBILE):
            return mobile_contacts;
    }
}